internal/apmhostutil/container_linux.go

```diff
@@ -99,32 +97,40 @@ func readCgroupContainerInfo(r io.Reader) (*model.Container, *model.Kubernetes,
 	var kubernetes *model.Kubernetes
 	s := bufio.NewScanner(r)
 	for s.Scan() {
+		// split the line according to the format "hierarchy-ID:controller-list:cgroup-path"
 		fields := strings.SplitN(s.Text(), ":", 3)
 		if len(fields) != 3 {
 			continue
 		}
+
+		// extract cgroup-path
 		cgroupPath := fields[2]
 
-		// Depending on the filesystem driver used for cgroup
-		// management, the paths in /proc/pid/cgroup will have
-		// one of the following formats in a Docker container:
-		//
-		//   systemd: /system.slice/docker-<container-ID>.scope
-		//   cgroupfs: /docker/<container-ID>
-		//
-		// In a Kubernetes pod, the cgroup path will look like:
-		//
-		//   systemd: /kubepods.slice/kubepods-<QoS-class>.slice/kubepods-<QoS-class>-pod<pod-UID>.slice/<container-iD>.scope
-		//   cgroupfs: /kubepods/<QoS-class>/pod<pod-UID>/<container-iD>
-		//
-		dir, id := path.Split(cgroupPath)
-		if strings.HasSuffix(id, systemdScopeSuffix) {
-			id = id[:len(id)-len(systemdScopeSuffix)]
-			if dash := strings.IndexRune(id, '-'); dash != -1 {
-				id = id[dash+1:]
+		// split based on the last occurrence of the colon character, if such exists, in order
+		// to support paths of containers created by containerd-cri, where the path part takes
+		// the form: <dirname>:cri-containerd:<container-ID>
+		idx := strings.LastIndex(cgroupPath, ":")
+		if idx == -1 {
+			// if colon char is not found within the path, the split is done based on the
+			// last occurrence of the slash character
+			if idx = strings.LastIndex(cgroupPath, "/"); idx == -1 {
+				continue
+			}
+		}
+
+		dirname, basename := cgroupPath[:idx], cgroupPath[idx+1:]
+
+		// If the basename ends with ".scope", check for a hyphen and remove everything up to
+		// and including that. This allows us to match .../docker-<container-id>.scope as well
+		// as .../<container-id>.
+		if strings.HasSuffix(basename, ".scope") {
```

Comment:
I think we can remove the `systemdScopeSuffix` constant?

Reply:
Good catch!